mission4/src/controllers/articlesController.js

```diff
@@ -0,0 +1,191 @@
+//게시글 목록 조회(좋아요 포함)
+export async function getArticleList(req, res) {
+  const { page, pageSize, orderBy, keyword } = create(req.query, GetArticleListParamsStruct);
+  const user = req.user;
+  const where = {
+    title: keyword ? { contains: keyword } : undefined,
+  };
+
+  const totalCount = await prisma.article.count({ where });
+  const articles = await prisma.article.findMany({
+    skip: (page - 1) * pageSize,
+    take: pageSize,
+    orderBy: orderBy === 'recent' ? { createdAt: 'desc' } : { id: 'asc' },
+    where,
+    include: user
+      ? {
+          likes: {
+            where: { userId: user.id },
+            select: { id: true },
+          },
+        }
+      : undefined,
+  });
+  const articlesWithIsLiked = articles.map((m) => {
+    const isLiked = user ? m.likes.length > 0 : false;
+    const { likes, ...articleWithoutLikes } = m;
+    return { ...articleWithoutLikes, isLiked };
+  });
+  return res.send({
+    list: articlesWithIsLiked,
+    totalCount,
+  });
+}
+//댓글 등록
+export async function createComment(req, res) {
+  const { id: articleId } = create(req.params, IdParamsStruct);
+  const { content } = create(req.body, CreateCommentBodyStruct);
+  const user = req.user;
+  await prisma.article.findUniqueOrThrow({ where: { id: articleId } });
+  const comment = await prisma.comment.create({
+    data: {
+      articleId,
+      content,
+      userId: user.id,
+    },
+  });
+
+  return res.status(201).send(comment);
+}
+//댓글 목록 조회
+export async function getCommentList(req, res) {
+  const { id: articleId } = create(req.params, IdParamsStruct);
+  const { cursor, limit } = create(req.query, GetCommentListParamsStruct);
+
+  const article = await prisma.article.findUnique({ where: { id: articleId } });
+  if (!article) {
+    throw new NotFoundError('article', articleId);
+  }
+
+  const commentsWithCursor = await prisma.comment.findMany({
+    cursor: cursor ? { id: cursor } : undefined,
+    take: limit + 1,
+    where: { articleId },
+    orderBy: { createdAt: 'desc' },
+  });
+  const comments = commentsWithCursor.slice(0, limit);
+  const cursorComment = commentsWithCursor[commentsWithCursor.length - 1];
+  const nextCursor = cursorComment ? cursorComment.id : null;
+
+  return res.send({
+    list: comments,
+    nextCursor,
+  });
+}
+//게시글 좋아요 등록
+export async function likeArticle(req, res) {
+  const { id: articleId } = create(req.params, IdParamsStruct);
+  const user = req.user;
+  if (!user) {
+    throw new UnauthorizedError();
+  }
+  const article = await prisma.article.findUniqueOrThrow({ where: { id: articleId } });
+  const existingLike = await prisma.like.findUnique({
+    where: {
+      userId_articleId: {
+        userId: user.id,
+        articleId,
+      },
+    },
+  });
+  if (existingLike) {
+    return res.status(400).send({ message: '해당 게시글에 이미 좋아요를 눌렀습니다.' });
+  }
+  await prisma.like.create({
+    data: {
+      userId: user.id,
+      articleId,
+      productId: null,
+    },
+  });
```

Comment:
Prisma에서 `optional` 필드는 명시하지 않으면 자동으로 `null`이 됩니다. `articleId: null` 또는 `productId: null`과 같이 명시적으로 작성하는 것은 불필요한 코드입니다.
```suggestion
await prisma.like.create({
  data: {
    userId: user.id,
    productId,
  },
});
```
이렇게 필요한 필드만 명시하면 더 간결하고 읽기 좋은 코드가 됩니다.